java/src/main/java/ai/rapids/cudf/ast/Operation.java

```diff
@@ -20,7 +20,7 @@
 import java.nio.ByteOrder;
 
 /** Base class of every AST expression. */
-public abstract class Expression extends AstNode {
+public abstract class Operation extends AstExpression {
   public CompiledExpression compile() {
```

Comment:
Now that all AST nodes can be an expression, this method should move to `AstExpression` because any expression can become a `CompiledExpression`.  I'm not sure this Java class still needs to exist afterwards.